### ISSUE TITLE
Aggregate on an annotated field errors out.

### ISSUE THREAD

**Issue by rohitkeshav** [mathebox/django_hana_pyhdb#11]
as_sql is called as .as_sql(with_col_aliases=True, subquery=True).
**TypeError: as_sql() got an unexpected keyword argument 'with_col_aliases**
because that parameter is not expected.
the as_sql method just seems to call the base class method.

**Comment by mathebox** [comment]
Hi @rohittk!
It seems django made some internal changes.
What python and django version are you using? What's the command you try to execute?

**Comment by rohitkeshav** [comment]
Hey @mathebox , 
python and django versions are 2.7 and 1.8.10 respectively. It was a normal query actually. annotated a field to count and the then tried summing the same annotated field using aggregate.

**Comment by amarvin** [comment]
I just got the same error with Python 3.5 and Django 1.10.3. The error was in line 214 of \django\db\models\sql\subqueries.py in the class AggregateQuery in the function add_subquery

**Comment by Sharad-em** [comment]
The SQLCompiler.as_sql template is: 
def as_sql(self, with_limits=True, with_col_aliases=False, subquery=False):

If this template can be followed and super method can be called with same signature, this issue can be fixed.

**Comment by amarvin** [comment]
In that case, is the issue in https://github.com/mathebox/django_hana_pyhdb/blob/157ab40c079e9075ef75de49808bf8129d51f4aa/django_hana/compiler.py then? Maybe in SQLInsertCompiler.as_sql.

**Comment by Sharad-em** [comment]
I think all as_sql methods should accept parameters declared by base class methods otherwise insert, updates will also fail. If these params are unused in the method, they should be simply passed on to the super call.

Here is the declaration of as_sql method in SQLCompiler class in django.db.models.sql.compiler (I am looking at django version 1.8.10)
   def as_sql(self, with_limits=True, with_col_aliases=False, subquery=False):

As one can see, the base implementation (the spec of this class) accepts 3 params but the derived implementation in django_hana/compiler.py does not. Which causes the issue.

I think as_sql method of django_hana/compiler.py should be changed to:

def as_sql(self, with_limits=True, with_col_aliases=False, subquery=False):
        result, params = super(SQLCompiler, self).as_sql(with_limits, with_col_aliases, subquery)
        update_params = self.connection.ops.modify_params(params)
        return result, update_params

**Comment by mathebox** [comment]
I would prefer the use of `args` and ` kwargs` to support API changes:
```python
def as_sql(self, *args, **kwargs):
    result, params = super(SQLCompiler, self).as_sql(*args, **kwargs)
    update_params = self.connection.ops.modify_params(params)
    return result, update_params
```

The same should be applied to `SQLInsertCompiler.as_sql` and `SQLUpdateCompiler.as_sql`.

**But** I would like to have one or multiple tests for this behavior. Currently I'm quite busy. Feel free to open a pull request.

**Comment by mathebox** [comment]
Fixed by #20
